src/expressServer.js

```diff
@@ -164,7 +162,7 @@ app.get('/tests', (req, res) => {
 // Error handling middleware (**MUST** be defined _after_ all other routes and middleware)
 app.use(notFoundMiddleware); // Not found middleware to handle invalid routes
 app.use(errorHandlerMiddleware); // Error handler middleware
-
+app.use('/api/v1/profile', imgur);
```

Comment:
Express reads the code in this file, from top to bottom. It will try to match the current URL to each line. You can explore this in the logs by enabling Express's debugger: [https://expressjs.com/en/guide/debugging.html](https://expressjs.com/en/guide/debugging.html) 

```
DEBUG=express:router npm run dev
```

This will add additional logging to the terminal to show what Express is doing. So now if you were to go to `localhost:8000/not_a_real_url.jpeg` in your terminal, you'll see that the request is handled by a bunch of middleware defined early in this file like `helmet`, then it eventually logs `express:router notFound  : /my_special_file.js +14ms` which means that it went top to bottom in this file, and the first route or middleware it found to handle this request that returns a response was `notFoundMiddleware`.

On the other hand, if we go to a route we know exists like `localhost:8000/api/v1/albums`, we see in the Express debug logs that the request goes through a bunch of middleware again, and then ends up printing `express:router router /api/v1/albums : /api/v1/albums +0ms` which shows that this route is being handled by the imported `albumRouter` on line 118. Since the route in the `albumRouter` file calls a controller function that returns a response `res.status(...).json(...)` that means Express knows it can return that response to the browser and it doesn't need to go through any of the rest of the routes or middlewares defined in this file like the `notFoundMiddleware` or the `errorHandlerMiddleware`

---

In your case, you have your `api/v1/profile` router defined _after_ the `notFoundMiddleware` and the `errorHandlerMiddleware`. That means Express is going to get to the `notFoundMiddleware` _**first.**_ The not-found middleware matches _any_ route and will return a 404 response. Once a response is defined, Express does not try to call any routes defined later in the file. 

To fix this, you should have your `app.use` router defitinition _before_ the `notFoundMiddleware`. This is the reason why we've grouped all of our routes together from line 116 - 122 in this file. And even our one-off routes like our `/health` route or our routes for EJS files or for API docs, all happen earlier in this file _before_ the `notFoundMiddleware`. 

**conclusion:** You should put the `app.use` line under the `chatRoutes` line (or anywhere in the file _before_ `notFoundMiddleware` is defined.

Reply:
Thank you very much , it is corrected , moved before the notFoundMiddleware is defined, commit d669580ef4c2171bb3476035eba87b46efdc642b